### ISSUE TITLE
"Eval" vs. "Train" distinction not captured in GenPyTorch

### ISSUE THREAD

**Issue by alex-lew** [probcomp/GenPyTorch.jl#3]
Torch has a distinction between "eval" and "train." Which mode you're in determines the behavior of layers like DropOut, which are meant to alter the training procedure but to be inactive at test time. 

A simple rule of thumb could be to treat `simulate` and `generate` as `eval`-mode operations, and `accumulate_param_gradients!` (and perhaps `choice_gradients`?) as `train`-mode. Or we could have it be a flag on the generative function itself (and make the generative function mutable, so the flag could be turned on and off).

**Comment by JGameCreation** [comment]
Can't you just use the PyTorch API to set the mode of the model, like `model.eval()` to set it to eval mode?
